### PR TITLE
Testcase for aggregating on content(204) responses at the aggregate m…

### PR DESCRIPTION
**Pull request by prabushi**
…ediator

## Purpose
Testcase for the fix https://wso2.org/jira/browse/ESBJAVA-4709

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/AggregateEmptyJsonPayloadTestCase.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ */
+public class AggregateEmptyJsonPayloadTestCase extends ESBIntegrationTest {
+    
+    private static final String PROXY_NAME = "aggregateEmptyJsonPayloadTestProxy";
+    private LogViewerClient logViewerClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
```
Comment (chanikag) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/AggregateEmptyJsonPayloadTestCase.java]: Don't we have to check the proxy service existence, before running the testcase?

**Comment by chanikag** [comment]
This depends on the fix https://github.com/wso2/wso2-synapse/pull/921. Need to release wso2-synapse and upgrade the synapse version before merging this PR

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/AggregateEmptyJsonPayloadTestCase.java (hunk 1 above)
Comment (prabushi) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/AggregateEmptyJsonPayloadTestCase.java]: Updated.

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/AggregateEmptyJsonPayloadTestCase.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * At the proxy service, requests will be sent iteratively for the no content response backend, and recieved
+     * responses will be aggregated
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb", description = "Test CorrelateOn in Aggregate mediator ")
+    public void testAggregateEmptyJsonPayload() throws Exception {
+
+        String inputPayload = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n" +
+                "<soapenv:Header/>\n" +
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/AggregateEmptyJsonPayloadTestCase.java]: Line wrapping should happen before an operator. Shall we correct in all places?

Comment (prabushi) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/AggregateEmptyJsonPayloadTestCase.java]: Updated.

**Pull request merged**